File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03802/p03802_num3_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<int> x(N), y(N);
    for (int i = 0; i < N; i++) {
        cin >> x[i] >> y[i];
    }
    
    int max_dist = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            int dist = max(abs(x[i] - x[j]), abs(y[i] - y[j]));
            max_dist = max(max_dist, dist);
        }
    }
    
    cout << max_dist << endl;
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03802/p03802_num3_parsed.py
import sys

def max_distance(N, coordinates):
    max_dist = 0
    for i in range(N):
        for j in range(i+1, N):
            dist = max(abs(coordinates[i][0]-coordinates[j][0]), abs(coordinates[i][1]-coordinates[j][1]))
            max_dist = max(max_dist, dist)
    return max_dist

if __name__ == "__main__":
    N = int(sys.stdin.readline().strip())
    coordinates = []
    for _ in range(N):
        x, y = map(int, sys.stdin.readline().strip().split())
        coordinates.append((x, y))
    print(max_distance(N, coordinates))